crates/net/common/src/metered_sender.rs

```diff
@@ -0,0 +1,64 @@
+//! Support for metering senders. Facilitates debugging by exposing metrics for number of messages
+//! sent, number of errors, etc.
+
+use metrics::Counter;
+use reth_metrics_derive::Metrics;
+use tokio::sync::mpsc::{
+    error::{SendError, TrySendError},
+    Sender,
+};
+
+/// Network throughput metrics
+#[derive(Metrics)]
+#[metrics(scope = "metered_sender")]
+struct MeteredSenderMetrics {
+    /// Number of messages sent
+    messages_sent: Counter,
+    /// Number of errors when sending messages
+    send_errors: Counter,
+}
+
+/// Manages updating the network throughput metrics for a metered stream
+pub struct MeteredSender<T> {
+    /// The [`Sender`] that this wraps around
+    sender: Sender<T>,
```

Comment:
Wonder if this can be made generic over any kind of Sink type? @mattsse. All good if not.

Reply:
hmm, I can't think of anything where this would be useful.

Reply:
sg, will keep it as-is for now and we can make the change later when there's demand for it